react/react2/week_1/github-user-searcher/src/SearchContext.jsx

```diff
@@ -0,0 +1,40 @@
+import { createContext, useEffect, useState } from "react";
+
+export const SearchingContext = createContext();
+
+export default function SearchProvider({ children }) {
+  const [query, setQuery] = useState("");
+  const [userList, setUserList] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const abortController = new AbortController();
+
+    setLoading(true);
+    fetch(`https://api.github.com/search/users?q=${query}`, {
+      signal: abortController.signal,
+    })
+      .then((response) => response.json())
+      .then((data) => setUserList(data.items))
+      .catch((error) => setError(error.message))
+      .finally(() => setLoading(false));
+
+      return () => {
+        abortController.abort();
+      };
+  }, [query]);
+
+  const handleInputChange = (e) => setQuery(e.target.value);
+
+  const renderUser = () =>
+    userList?.map((user) => <p key={user.id}>{user.login}</p>);
```

Comment:
Rendering data should not be used in the Context Provider as it is to share data globally. [for your reference](https://legacy.reactjs.org/docs/context.html)

Reply:
Thank you for your feedback and the reference! 
I have fixed my code, Do you mind to confirm me if I need to adjust sth else or I can merge?